src/Sarif.Viewer.VisualStudio/Sarif/Replacement.extensions.cs

```diff
@@ -19,15 +19,18 @@ public static ReplacementModel ToReplacementModel(this Replacement replacement)
 
             ReplacementModel model = new ReplacementModel();
 
-            model.DeletedLength = replacement.DeletedLength;
+            model.DeletedLength = replacement.DeletedRegion.Length;
+            model.Offset = replacement.DeletedRegion.Offset;
 
-            if (!String.IsNullOrEmpty(replacement.InsertedBytes))
+            if (!string.IsNullOrWhiteSpace(replacement.InsertedContent?.Text))
             {
-                model.InsertedString = Encoding.UTF8.GetString(Convert.FromBase64String(replacement.InsertedBytes));
-                model.InsertedBytes = Encoding.UTF8.GetBytes(model.InsertedString);
+                model.InsertedString = replacement.InsertedContent.Text;
+            }
+            else if (replacement.InsertedContent?.Binary != null)
+            {
+                model.InsertedBytes = Convert.FromBase64String(replacement.InsertedContent.Binary);
+                model.InsertedString = Encoding.UTF8.GetString(model.InsertedBytes);
```

Comment:
If the content is expressed in binary, there's no guarantee that decoding it into a string is sensible. And if it is, you don't know the encoding. If you wanted to be fancy you could check for the file's encoding (I don't know if you have access to `file.encoding` here), but otherwise, you can't populate `InsertedString` (and the rest of the Viewer needs to be ok with it being absent).

Reply:
Understood, fixed.